src/avx512bw/avx512bw-from-utf8.inl.cpp

```diff
@@ -0,0 +1,462 @@
+/*
+validate_utf8_structure only checks if leading bytes
+are follewed by proper number of continuation bytes
+and then there's another leading byte.
+
+We don't handle end-of-stream here.
+*/
+bool validate_utf8_structure(__m512i input) {
+    /* 1. bitmask for various character byte classes.
+
+        leading: 111010011100011010001
+                 abccdddefgggghiijjjjk   a-k -- 11 UTF8- chars characters
+
+        ascii:   110000011000010000001
+                 ab     ef    h      k
+
+        2 bytes: 001000000000001000000
+                   cc          ii
+
+        3 bytes: 000010000000000000000
+                     ddd              
+
+        4 bytes: 000000000100010010000
+                          gggg   jjjj
+    */
+    uint64_t leading;
```

Comment:
Possibly, these should be '__mmask64'. Be mindful that it is 3 cycles to go from __mmask64 to uint64_t. The compiler might be smart about it, but...

Reply:
True. Currently. GCC cannot handle well the mixing of bitops with kmask and regular words. It always cast kmasks into words. Some time ago a filled a bug for this: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=88798.